templates/test/dev/custom-builds/kustomization.yaml

```
@@ -0,0 +1,6 @@
+namespace: default
+resources:
+  - ../../../flavors/base
+  - machine-deployment.yaml
```

we should be able to reuse ../../../flavors/default

```suggestion
  - ../../../flavors/default
```

---

I haven't tested with that change. What's the difference?

---

[`.spec.template.spec.location`](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/066f8648469aee8a037bcec4dcb3e2abb810be7f/templates/test/dev/custom-builds/machine-deployment.yaml#L32) was removed in v1alpha4

```diff
diff --git a/templates/test/dev/cluster-template-custom-builds.yaml b/templates/test/dev/cluster-template-custom-builds.yaml
index 0a07cc83..1a02cf9b 100644
--- a/templates/test/dev/cluster-template-custom-builds.yaml
+++ b/templates/test/dev/cluster-template-custom-builds.yaml
@@ -232,7 +232,6 @@ metadata:
 spec:
   template:
     spec:
-      location: ${AZURE_LOCATION}
       osDisk:
         diskSizeGB: 128
         managedDisk:
@@ -305,4 +304,3 @@ spec:
           name: '{{ ds.meta_data["local_hostname"] }}'
       postKubeadmCommands:
       - bash -c /tmp/replace-k8s-components.sh
-      useExperimentalRetryJoin: true
diff --git a/templates/test/dev/custom-builds/kustomization.yaml b/templates/test/dev/custom-builds/kustomization.yaml
index 2dbfb95b..9187d069 100644
--- a/templates/test/dev/custom-builds/kustomization.yaml
+++ b/templates/test/dev/custom-builds/kustomization.yaml
@@ -1,6 +1,5 @@
 namespace: default
 resources:
-  - ../../../flavors/base
-  - machine-deployment.yaml
+  - ../../../flavors/default
 patchesStrategicMerge:
   - patches/custom-builds.yaml

```

---

+1 let's use default instead duplicating the machine-deployment.yaml file 

I'm guessing these values have changed since you opened this PR in the file it was copied from  https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/templates/flavors/default/machine-deployment.yaml

---

@jackfrancis this still needs to be addressed

---

Thanks, updated so that we aren't copy/pasting the `machine-deployment.yaml` and are using default instead of base.